docs/design/deviceshifu/design-deviceShifu-plc4x.md

```diff
@@ -0,0 +1,70 @@
+# DeviceShifu PLC4X Design
+
+DeviceShifu-PLC4X allows shifu utilize [apache plc4x](https://plc4x.apache.org/) project to integrate PLC devices with various protocols such as S7, Modbus, etc.
+
+## Goal
+Create a deviceShifu-PLC4X type using PLC4X library and allow user to connect PLC devices easily
+
+## General Design
+
+deviceShifu-PLC4X will receive RESTful style request like other, then transfer the request into valid PLC4X method calls and send it to connected devices.
+
+## Detailed Design
+
+### Protocol Specification
+PLC4X can support various protocols, thus when initializing deviceShifu-PLC4X, user should specify which protocol to use. In order to support protocol specification,
+we can add new settings to edgedevice_types.go to support it.
+
+```go
+type ProtocolSettings struct {
+	MQTTSetting   *MQTTSetting   `json:"MQTTSetting,omitempty"`
+	OPCUASetting  *OPCUASetting  `json:"OPCUASetting,omitempty"`
+	SocketSetting *SocketSetting `json:"SocketSetting,omitempty"`
++	PLC4XSetting  *PLC4XSetting  `json:"PLC4XSetting,omitempty"`
+}
+
+// PLC4XSetting defines PLC4X specific settings when connecting to an EdgeDevice
+type PLC4XSetting struct {
+    DriverType *string `json:"driverType,omitempty"`
+}
+```
+DeviceShifu-PLC4X will maintain an enum of protocols and match the driver base on the setting:
+```go
+ swtich plc4xSetting.DrvierType:
+	 case S7:
+		 drviers.RegsiterS7(drvierManager)
+		 ...
+```
+
+### Serving requests
+deviceShifu-PLC4X would take RESTful-style requests just as other deviceShifu do. 
+PLC4X supports both `read` and `write` requests. 
+
+For read, the method signature looks like:
```

Comment:
Do we need to support multiple requests being sent at the same time? like AddQuery().AddQuery()....